### DISCUSSION TITLE
SBT packages fail to update

### DISCUSSION THREAD

**Discussion by gwendolyngoetz** [renovatebot/renovate#27850] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Previous issue [here](https://github.com/renovatebot/renovate/issues/27653)

I'm still having errors which repro both in our project and my [minimal example repo](https://github.com/gwendolyngoetz/renovate-sbt-error).

Running the cli `platform=local` succeeds, but when run normally is where the failure occurs.

I get this warning: "Package lookup failures" and this is shown on the dashboard "Failed to look up github-releases package sbt/sbt`". There tend to be a lot of http 405s or http 404s to maven.org, scala-sbt.org, and shibboleth.net.

```json
{
  "warnings": [
    "Failed to look up github-releases package sbt/sbt"
  ]
  "files": [
    "server/project/build.properties"
  ]
}
```

Further back in the log output there is a POST to maven with an http 405 result.

It was commented in the other issue that the [wrong url for sbt is being used](https://github.com/renovatebot/renovate/issues/27653#issuecomment-1987673969).

Here are full log files for the minimal repro and my full project
[sample-repo-logs.txt](https://github.com/renovatebot/renovate/files/14552539/sample-repo-logs.txt)
[civiform-repo-logs.txt](https://github.com/renovatebot/renovate/files/14552547/civiform-logs.txt)

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: hostRules: no authentication for repo1.maven.org
DEBUG: Using throttle 200 intervalMs for host repo1.maven.org
DEBUG: Using queue: host=repo1.maven.org, concurrency=5
DEBUG: POST https://repo1.maven.org/maven2/api/graphql = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=405 retryCount=0, duration=24)
DEBUG: Datasource connection error
{
  "datasource": "github-releases"
  "packageName": "sbt/sbt"
  "url": "https://repo1.maven.org/maven2/api/graphql"
  "errCode": "ERR_NON_2XX_3XX_RESPONSE"
}

DEBUG: Failed to look up github-releases package sbt/sbt
{
  "dependency": "sbt/sbt"
  "packageFile": "server/project/build.properties"
}

DEBUG: PackageFiles.add() - Package file saved for base branch
{
  "baseBranch": "master"
}

DEBUG: Package releases lookups complete
{
  "baseBranch": "master"
}

```

</details>

**Comment by rarkins** [comment]
An incorrect registryUrls seem to be present, possibly from the sbt extract process. Reproduction result:

```
DEBUG: packageFiles with updates
{
  "baseBranch": "master"
  "config": {
    "sbt": [
      {
        "deps": [
          {
            "datasource": "github-releases",
            "depName": "sbt/sbt",
            "packageName": "sbt/sbt",
            "versioning": "semver",
            "currentValue": "1.9.9",
            "replaceString": "sbt.version=1.9.9",
            "extractVersion": "^v(?<version>\\S+)",
            "registryUrls": [
              "https://repo1.maven.org/maven2"
            ],
            "updates": [],
            "warnings": [
              {
                "topic": "sbt/sbt",
                "message": "Failed to look up github-releases package sbt/sbt"
              }
            ]
          }
        ],
        "packageFile": "server/project/build.properties"
      }
    ]
  }
}
```

**Reply by viceice** [reply]
I think it's a regression because of the recent registry url handling changes in manager.

**Reply by Churro** [reply]
Most likely introduced with https://github.com/renovatebot/renovate/pull/27276:
https://github.com/renovatebot/renovate/blob/d111a1e80e71d3e3fac881ff75d004881cff4fce/lib/modules/manager/sbt/extract.ts#L396-L406

Previously `registryUrls` was `[]` for deps with `GithubReleasesDatasource`, now `https://repo1.maven.org/maven2` is set. In other words, the functionality to update `sbt/sbt` is broken.

**Reply by lpriyasivakumar** [reply]
@Churro @viceice Raised a pr and updated the test here https://github.com/renovatebot/renovate/pull/27856
Lmk if you have feedback. Thank you
